### PR TITLE
added yarnv3 nagger deprication warning

### PR DESCRIPTION
**Pull request by narinder17**
Resolves # . (This is applicable only if this pull request relates to a GitHub issue, delete the line otherwise)

[dtspo-10794](https://tools.hmcts.net/jira/browse/DTSPO-10974) - adding a nagger warning to advise users to upgrade to yarnV3.

### CODE REVIEW

File: src/uk/gov/hmcts/contino/YarnBuilder.groovy [hunk 1]
```diff
@@ -308,6 +308,12 @@ EOF
     return status
   }
 
+  private yarnDepreciationCheck() {
+     if (!isYarnV2OrNewer()){
+       WarningCollector.addPipelineWarning("packageManager required", "Please upgrade to Yarn V3 by adding packageManager to package.json. For examples see: https://github.com/hmcts/cnp-plum-frontend/blob/2bfb6eac580c0965e331731496619c0b5920d596/package.json#L47 and https://github.com/hmcts/dtsse-ardoq-adapter", LocalDate.of(2023, 03, 10 ))
```
Comment (timja) [inline src/uk/gov/hmcts/contino/YarnBuilder.groovy:313]: better if you can send a pull request suggestion for upgrading or pointing to the slack announcement  https://moj.enterprise.slack.com/files/T1L0WSW9F/F04784SLAJC?origin_team=T1L0WSW9F
```suggestion
       WarningCollector.addPipelineWarning("old_yarn_version", "Please upgrade to Yarn V3 by adding packageManager to package.json. For examples see: https://github.com/hmcts/cnp-plum-frontend/blob/2bfb6eac580c0965e331731496619c0b5920d596/package.json#L47 and https://github.com/hmcts/dtsse-ardoq-adapter", LocalDate.of(2023, 03, 10 ))
```

File: src/uk/gov/hmcts/contino/YarnBuilder.groovy [hunk 2]
```diff
@@ -308,6 +308,12 @@ EOF
     return status
   }
 
+  private yarnDepreciationCheck() {
+     if (!isYarnV2OrNewer()){
+       WarningCollector.addPipelineWarning("old_yarn_version", "Please upgrade to Yarn V3 by adding packageManager to package.json. For examples see: https://github.com/hmcts/cnp-plum-frontend/blob/2bfb6eac580c0965e331731496619c0b5920d596/package.json#L47 and https://github.com/hmcts/dtsse-ardoq-adapter", LocalDate.of(2023, 03, 10 ))
```
Comment (timja) [inline src/uk/gov/hmcts/contino/YarnBuilder.groovy:313]: Please improve the message, e.g
```suggestion
       WarningCollector.addPipelineWarning("old_yarn_version", "Please upgrade to Yarn V3, see https://moj.enterprise.slack.com/files/T1L0WSW9F/F04784SLAJC?origin_team=T1L0WSW9F", LocalDate.of(2023, 03, 10 ))
```

**Review by timja — APPROVED** [review]

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+    "java.configuration.updateBuildConfiguration": "interactive"
+}
```
Comment (timja) [inline .vscode/settings.json:3]: Please remove

Comment (narinder17) [reply .vscode/settings.json]: folder removed as requested.

File: src/uk/gov/hmcts/contino/YarnBuilder.groovy [hunk 4]
```diff
@@ -308,6 +308,12 @@ EOF
     return status
   }
 
+  private yarnDepreciationCheck() {
```
Comment (timja) [inline src/uk/gov/hmcts/contino/YarnBuilder.groovy:311]:
```suggestion
  private nagAboutOldYarnVersions() {
```

**Review by timja — CHANGES_REQUESTED** [review]
you need to call the nagger from somewhere. Please add links to where this has been tested and any slack messages

File: src/uk/gov/hmcts/contino/YarnBuilder.groovy [hunk 5]
```diff
@@ -336,6 +342,7 @@ EOF
   @Override
   def setupToolVersion() {
     super.setupToolVersion()
+
```
Comment (timja) [inline src/uk/gov/hmcts/contino/YarnBuilder.groovy:345]:
```suggestion
    super.setupToolVersion()
    nagAboutOldYarnVersions()

```

**Review by timja — COMMENTED** [review]
Links to testing? Including builds and slack messages.

(FYI I'm going on leave for two weeks so feel free to get someone else to approve if all is working, the code looks good to me)

**Comment by narinder17** [comment]
The nagger error messages appears and runs fine in Jenkins library, shows error in cnp-plum-frontend repo. The build fails due to other dependancies in the cnp-plum-frontend repo, however this branch will not be merged and was used to point at https://github.com/hmcts/cnp-jenkins-library/pull/967 build.

https://tools.hmcts.net/jira/browse/DTSPO-10974

image

image

Please upgrade to Yarn V3, see [https://moj.enterprise.slack.com/files/T1L0WSW9F/F04784SLAJC?origin_team=T1L0WSW9F](https://moj.enterprise.slack.com/files/T1L0WSW9F/F04784SLAJC?origin_team=T1L0WSW9F&origin_channel=D04KECS7690) This configuration will stop working by 10/03/2023 ( in 44 days )

**Review by thomast1906 — APPROVED** [review]

**Review by thomast1906 — APPROVED** [review]

**Pull request merged**